apps/demos/Demos/Scheduler/ContextMenu/React/App.tsx

```diff
@@ -1,19 +1,22 @@
 import React, { useCallback, useRef, useState } from 'react';
+
 import Scheduler, { Resource, type SchedulerTypes, SchedulerRef } from 'devextreme-react/scheduler';
 import ContextMenu, { type ContextMenuTypes } from 'devextreme-react/context-menu';
-import { data, resourcesData, Resource as ResourceItem } from './data.ts';
+
+import { data, resourcesData } from './data.ts';
+import type { Resource as ResourceItem } from './types';
 import AppointmentMenuTemplate from './AppointmentTemplate.tsx';
 
 // eslint-disable-next-line no-unused-vars
-type ContextMenuItem = ContextMenuTypes.Item & ResourceItem & { onItemClick?: (e: ContextMenuTypes.ItemClickEvent) => void };
+type ContextMenuItem = ContextMenuTypes.Item & ResourceItem & { onItemClick?: (e: ContextMenuTypes.ItemClickEvent<ContextMenuItem>) => void };
```

Comment:
Let's move this type to data.ts as all other types
It is not clear, whether we need extension of ContextMenuTypes.Item here, looks like custom type is enough here 
```suggestion
export type ResourceItem = {
  id?: number;
  text: string;
  color?: string;
  onItemClick?: (e: ContextMenuTypes.ItemClickEvent<ResourceItem>) => void
}
```

Reply:
moved it to types.ts
but left Resource type as it was, because it is the type of dataSource for Resource component
left this type as is for now because some context menu items have beginGroup and disabled properties, which are defined in ContextMenuTypes.Item